### PR TITLE
[2/n] Interop Stack: Use AssetJobInfo to construct ExternalAssetNodes

### PR DESCRIPTION
**Pull request by OwenKephart**
Now that we have job-level asset info available to us, use that rather than iterating over every single OutputDefinition.

This is mostly an invisible change (which prepares us for a world in which this job level asset info is calculated without OutputDefinitions), but the major consequence is that I've decided to stop populating the input_name/output_name fields on the ExternalAssetDependency/ExternalAssetDependedBy objects.

The main reason for this is that these values are pretty weird to reason about once we get into graph land. While the output_name is fairly straightforward, the input_name for an asset dependency makes some assumptions that make it hard to calculate when an asset is backed by an op. 

It doesn't seem like these fields are really used anywhere in the UI, so keeping them None seems to be reasonable, but if that's not the case, I can figure something else out.

### CODE REVIEW

**Comment by OwenKephart** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7473** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7473" target="_blank">Graphite</a> 
    * **PR #7475** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7475" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/7475?utm_source=stack-comment).

File: python_modules/dagster/dagster/core/definitions/assets_info.py [hunk 1]
```diff
@@ -166,6 +166,14 @@ def from_graph(graph_def: GraphDefinition) -> "AssetsJobInfo":
             asset_deps=asset_deps,
         )
 
+    @property
+    def asset_info_by_node_input_handle(self) -> Mapping[NodeInputHandle, AssetInfo]:
```
Comment (sryza) [inline python_modules/dagster/dagster/core/definitions/assets_info.py:170]: Prefer `asset_infos_by_node_input_handle`, unless you'd argue that "info" is an [uncountable noun](https://www.englishclub.com/grammar/nouns-countable-un.htm) and a list of AssetInfo objects should be called `asset_info`?

Comment (OwenKephart) [reply python_modules/dagster/dagster/core/definitions/assets_info.py:170]: yeah the word "infos" was kinda unaesthetic to me. i don't have strong feelings about it though so I can change it if you feel less ambivalent than I do

**Review by sryza — APPROVED** [review]
<3 all the removed code

**Pull request merged**